Lib/test/test_importlib/extension/test_loader.py

```diff
@@ -29,6 +29,28 @@ def test_load_module_API(self):
         with self.assertRaises(ImportError):
             self.load_module('XXX')
 
+    def test_type_error(self):
+        with self.assertRaises(TypeError) as cm:
+            other = self.machinery.ExtensionFileLoader(
+                fullname=util.EXTENSIONS.name,
+                name=util.EXTENSIONS.name,
+                path=util.EXTENSIONS.file_path)
+
+        self.assertEqual(
+            str(cm.exception),
+            'Either name or fullname needs to be used'
```

Comment:
We typically discourage checking exact exception wording as it tends to be brittle. I would just check that `name` and `fullname` are in the message.